test/unit/metadata_test.py

```diff
@@ -39,3 +42,60 @@ def test_get_etl_group_tables() -> None:
     """Test that a Value error is raised for non existent etl group."""
     with pytest.raises(ValueError):
         Package.get_etl_group_tables("not_an_etl_group")
+
+
+@pytest.mark.parametrize("resource_name", sorted(PUDL_RESOURCES.keys()))
+def test_pyarrow_schemas(resource_name: str):
+    """Verify that we can produce pyarrow schemas for all defined Resources."""
+    _ = PUDL_RESOURCES[resource_name].to_pyarrow()
+
+
+@pytest.mark.parametrize("field_name", sorted(FIELD_METADATA.keys()))
+def test_field_definitions(field_name: str):
+    """Check that all defined fields are valid."""
+    _ = Field(name=field_name, **FIELD_METADATA[field_name])
+
+
+@pytest.mark.xfail(reason="Need to purge unused fields. See issue #3224")
+def test_defined_fields_are_used():
+    """Check that all fields which are defined are actually used."""
+    used_fields = set()
+    for resource in PUDL_RESOURCES.values():
+        used_fields |= {f.name for f in resource.schema.fields}
+    defined_fields = set(FIELD_METADATA.keys())
+    unused_fields = sorted(defined_fields - used_fields)
+    if len(unused_fields) > 0:
+        raise AssertionError(
+            f"Found {len(unused_fields)} unused fields: {unused_fields}"
+        )
```

Comment:
I don't *think* there are good reasons to have unused fields laying around.  I suspect that the ones we've got are just leftovers where we renamed a column and never purged the old definition.